pkg/controllers/internalmembercluster/v1beta1/member_controller_integration_test.go

```diff
@@ -2,220 +2,834 @@
+							},
+						},
+					},
+				}
+
+				if diff := cmp.Diff(
+					imc.Status, wantIMCStatus,
+					ignoreAllTimeFields,
+					sortByConditionType,
+				); diff != "" {
+					return fmt.Errorf("InternalMemberCluster status diff (-got, +want):\n%s", diff)
+				}
+
+				// Verify the last transition timestamps in the conditions.
+				//
+				// Note that at this point the structure of the InternalMemberCluster status
+				// object is already known.
+				conds := imc.Status.AgentStatus[0].Conditions
+				for idx := range conds {
+					cond := conds[idx]
+					if cond.LastTransitionTime.Before(&timeStarted) {
+						return fmt.Errorf("InternalMemberCluster agent status condition %s has last transition time %v, want before %v", cond.Type, cond.LastTransitionTime, timeStarted)
+					}
+				}
+
+				// Verify the observation timestamps in the properties.
+				for pn, pv := range imc.Status.Properties {
+					if pv.ObservationTime.Before(&timeUpdated) {
+						return fmt.Errorf("InternalMemberCluster property %s has observation time %v, want before %v", pn, pv.ObservationTime, timeStarted)
+					}
+				}
+
+				if u := imc.Status.ResourceUsage; u.ObservationTime.Before(&timeUpdated) {
+					return fmt.Errorf("InternalMemberCluster resource usage has observation time %v, want before %v", u.ObservationTime, timeStarted)
+				}
+
+				return nil
+			}, eventuallyTimeout, eventuallyInterval).Should(Succeed(), "Failed to update the agent status")
+		})
+
+		It("can mark the cluster as left", func() {
+			// Add an offset of -1 second to avoid flakiness caused by approximation.
+			timeLeft = metav1.Time{Time: time.Now().Add(-time.Second)}
+
+			imc := &clusterv1beta1.InternalMemberCluster{}
+			objKey := types.NamespacedName{
+				Name:      member2Name,
+				Namespace: member2ReservedNSName,
+			}
+			Expect(hubClient.Get(ctx, objKey, imc)).Should(Succeed())
+
+			imc.Spec.State = clusterv1beta1.ClusterStateLeave
+			Expect(hubClient.Update(ctx, imc)).Should(Succeed())
+		})
+
+		It("should let the cluster go", func() {
+			Eventually(func() error {
+				imc := &clusterv1beta1.InternalMemberCluster{}
+				objKey := types.NamespacedName{
+					Name:      member2Name,
+					Namespace: member2ReservedNSName,
+				}
+				if err := hubClient.Get(ctx, objKey, imc); err != nil {
+					return fmt.Errorf("failed to get InternalMemberCluster: %w", err)
+				}
+
+				wantIMCStatus := clusterv1beta1.InternalMemberClusterStatus{
+					Properties: map[clusterv1beta1.PropertyName]clusterv1beta1.PropertyValue{
+						propertyprovider.NodeCountProperty: {
+							Value: "3",
+						},
+					},
+					ResourceUsage: clusterv1beta1.ResourceUsage{
+						Capacity: corev1.ResourceList{
+							corev1.ResourceCPU:    resource.MustParse("15"),
+							corev1.ResourceMemory: resource.MustParse("49Gi"),
+						},
+						Allocatable: corev1.ResourceList{
+							corev1.ResourceCPU:    resource.MustParse("13"),
+							corev1.ResourceMemory: resource.MustParse("43Gi"),
+						},
```

Comment:
are those value tied to how we setup the integration test?

Reply:
Yeah, I will link them to common variables for better simplicity.